src/VisualStudio/CSharp/Impl/LanguageService/CSharpCodeCleanupFixer.cs

```diff
@@ -0,0 +1,157 @@
+                                                                           IDEDiagnosticIds.InlineDeclarationDiagnosticId,
+                                                                           CSharpRemoveUnusedVariableCodeFixProvider.CS0168,
+                                                                           CSharpRemoveUnusedVariableCodeFixProvider.CS0219,
+                                                                           IDEDiagnosticIds.UseObjectInitializerDiagnosticId,
+                                                                           IDEDiagnosticIds.UseCollectionInitializerDiagnosticId);
+        
+        public CSharpCodeCleanupFixer(ICodeFixService codeFixService, IThreadingContext threadingContext, bool assertIsForeground = false)
+            : base(threadingContext, assertIsForeground)
+        {
+            _codeFixServiceOpt = codeFixService;
+        }
+
+        public async Task<bool> FixAsync(ICodeCleanUpScope scope, FixIdContainer enabledFixIds, CancellationToken cancellationToken)
+        {
+            var textBufferScope = scope as TextBufferCodeCleanUpScope;
+            if (textBufferScope == null)
+            {
+                var hierarchyContent = scope as IVsHierarchyCodeCleanupScope;
+                var hierarchy = hierarchyContent.Hierarchy;
+                var itemId = hierarchyContent.ItemId;
+
+                if (hierarchy.GetCanonicalName(itemId, out var path) == 0)
+                {
+                    var attr = File.GetAttributes(path);
+                    if (attr.HasFlag(FileAttributes.Directory))
+                    {
+                        // directory
+                    }
+                    else
+                    {   
+                        // document
+                    }
+                }
+
+            }
+            else
+            {
+                var buffer = textBufferScope.SubjectBuffer;
+                if (buffer != null)
+                {
+                    var document = buffer.CurrentSnapshot.GetOpenDocumentInCurrentContextWithChanges();
+                    await CleanupDocument(document, cancellationToken).ConfigureAwait(false);
+                }
+            }
+            return false;
+        }
+
+        private async Task<bool> CleanupDocument(Document document, CancellationToken cancellationToken)
+        {
+            var oldDoc = document;
+            
+            foreach (var errorCode in _errorCodes)
+            {
+                var changedDoc = await _codeFixServiceOpt.ApplyCodeFixesForSpecificDiagnosticId(document, errorCode, cancellationToken).ConfigureAwait(true);
+                if (changedDoc != null)
+                {
+                    document = changedDoc;                    
+                }
+            }
+
+            document = await RemoveAndSortUsingsAsync(document, cancellationToken).ConfigureAwait(true);
```

Comment:
this seems duplicates of code clean up code? can we refactor to have only 1 implementation?

Reply:
The code clean up implementation is getting removed as part of a separate pull request for 16.0 preview 1.